File: src/main/java/skywolf46/extrautility/events/PlayerDamagedByEntityEvent.java
package skywolf46.extrautility.events;

import lombok.Getter;
import lombok.Setter;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;
import org.bukkit.event.Cancellable;
import org.bukkit.event.HandlerList;
import org.bukkit.event.player.PlayerEvent;

public class PlayerDamagedByEntityEvent extends PlayerEvent implements Cancellable {
    private static HandlerList handlerList = new HandlerList();
    @Getter
    private Entity damager;
    @Getter
    @Setter
    private double damage;
    @Getter
    @Setter
    private boolean cancelled;

    public PlayerDamagedByEntityEvent(Player who, Entity damager, double dmg, boolean isCancelled) {
        super(who);
        this.damager = damager;
        this.damage = dmg;
        this.cancelled = isCancelled;
    }

    @Override
    public HandlerList getHandlers() {
        return handlerList;
    }

    public static HandlerList getHandlerList() {
        return handlerList;
    }

}

File: src/main/kotlin/skywolf46/extrautility/events/combat/PlayerDamagedByEntityEvent.kt
package skywolf46.extrautility.events.combat

import org.bukkit.entity.Entity
import org.bukkit.entity.Player
import org.bukkit.event.Cancellable
import org.bukkit.event.HandlerList
import org.bukkit.event.player.PlayerEvent

class PlayerDamagedByEntityEvent(
    who: Player?,
    val damager: Entity,
    var damage: Double,
    private var c: Boolean,
) :
    PlayerEvent(who), Cancellable {

    private var cancelled: Boolean = false

    init {
        cancelled = c
    }

    override fun setCancelled(p0: Boolean) {
        cancelled = p0
    }

    override fun isCancelled(): Boolean = cancelled

    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        val handlerList = HandlerList()
    }

}
